Added DuplicateKeyError exception handling for Bad Request

Akshay-gaikwad opened the pull request:
Raised `APIException` when we get `DuplicateKeyError`
while trying to store duplicate event in database.

Akshay-gaikwad committed:
Added DuplicateKeyError exception handling for Bad Request

Raised APIException when we get DuplicateKeyError
while trying to store duplicate event in database.

almonteb commented on tycho/routes/event.py:
```diff
@@ -139,9 +140,12 @@ async def put_event(request, event: Event) -> Event:
-        LOG.info(json.dumps(event.to_primitive()))
+    try:
+        await request.app["db"].event.save(event)
+        if request.app["config"].log_events:
```
should this log statement be within the try? it doesn't seem like it could cause a `DuplicateKeyError`

Akshay-gaikwad replied:
Moved log statement outside try.

almonteb commented on tycho/routes/event.py:
```diff
@@ -139,9 +140,12 @@ async def put_event(request, event: Event) -> Event:
+        if request.app["config"].log_events:
+            LOG.info(json.dumps(event.to_primitive()))
+    except DuplicateKeyError as err:
+        raise APIException(message=f"Failed to add event. Error says: {err}")
```
should we be a bit more specific here? something along the lines of: "Error ID already exists"?

we may want to consider not leaking the exception to the client since it may include internal implementation details of how tycho is built and stores data.

Akshay-gaikwad replied:
Updated error message for DuplicateKeyError.

Akshay-gaikwad committed:
Updated error message for DuplicateKeyError

Updated error message for DuplicateKeyError
and moved log statement outside try
